utilities/operations/objectType/getOneObjectType.js

```diff
@@ -112,6 +115,34 @@ const getWobjWithFilters = async ({
   return { wobjects };
 };
 
+// check the ability to reserve this campaign
+const campaignValidation = async (campaigns) => {
+  const validCapaigns = [];
+  await Promise.all(campaigns.map(async (campaign) => {
+    if (campaign.reservation_timetable[moment().format('dddd').toLowerCase()]
+        && _.floor(campaign.budget / campaign.reward) > _.filter(campaign.users, (user) => user.status === 'assigned'
+        && user.createdAt > moment().startOf('month')).length) {
```

Comment:
Move if condition to some separated validate method, rename campaignValidation, because it's not validation, it's filter method.